### DISCUSSION TITLE
Remove field from submission before emailing

### DISCUSSION THREAD

**Discussion by earthwytch** [statamic/cms#8567] [Q&A] [answered]
I have a custom SendMail class that does some footling to submissions before they are emailed to recipients. This is because the submissions are emailed to our D365 CRM and processed automatically. Our D365 vendor is painfully expensive and slow to engage, so where possible, we need to make the CRM think we're still sending the same data even though we're adding a few fields here and there 💃 

This is a basic proof of concept in the handle method:

**/app/Forms/MySendMail.php**
```
public function handle()
{

    // before we send a submission, we need to check if it is a special snowflake
    $formHandle = $this->submission->form()->handle();

    if ( $formHandle == "account_change" ) {

        // add extra ContactPerson field to the existing Message field
        $concatenatedMessage = $this->submission->get('Message') . " " . $this->submission->get('ContactPerson');
        $this->submission->set('Message', $concatenatedMessage);

        // remove the ContactPerson field from the Submission so it doesn't show up in the email to the CRM
        ??
    }

    Mail::send(new Email($this->submission, $this->config, $this->site));
}
```

Is there a way to remove the extra field from the Submission before I give it to the Mail::send method? I can remove the extra field at the email template end, but I want to confine this disaster to one spot if possible.

**Comment by jasonvarga** [comment, marked as answer]
You should be able to do `$submission->remove($key)`

**Reply by earthwytch** [reply]
That worked to remove it from the submission 👍 thanks!

When it gets to the email view it has pulled it back in (Email::class -> addData()) and is still displaying it unfortunately so I'll hunt for another approach, though realistically with deadlines approaching it is probably simpler just to remove it with an if statement in the email view for now.
